src/composables/websocket.ts

```diff
@@ -0,0 +1,172 @@
+import { v4 as genUuid } from 'uuid';
+import { ref, watch } from '@vue/composition-api';
+import camelcaseKeysDeep from 'camelcase-keys-deep';
+import { handleUnknownError } from '../popup/utils';
+import type {
+  IDefaultComposableOptions,
+  IMiddlewareWebSocketSubscriptionMessage,
+  INetwork,
+  WebSocketChannelTypes,
+} from '../types';
+import { useGetter } from './vuex';
+
+let wsClient : WebSocket;
+const isWsConnected = ref(false);
+// any should be callback
+const subscribers: Record<WebSocketChannelTypes, Record<string, any>> = {
+  Transactions: {},
+  MicroBlocks: {},
+  KeyBlocks: {},
+};
+
+const channels: Record<WebSocketChannelTypes, boolean> = {
+  Transactions: false,
+  MicroBlocks: false,
+  KeyBlocks: false,
+};
+
+// eslint-disable-next-line no-unused-vars
+export function useWebSocket({ store }: IDefaultComposableOptions) {
+  const activeNetwork = useGetter<INetwork>('activeNetwork');
+
+  /**
+   * This function is used to subscribe to a specific WebSocket channel.
+   * @param {WebSocketChannelTypes} channelName - The name of the channel to subscribe to
+   */
+  function subscribeToWebSocketChannel(channelName: WebSocketChannelTypes) {
+    if (
+      !channelName
+      || channels[channelName]
+      || !isWsConnected.value
+    ) {
+      return;
+    }
+
+    wsClient.send((JSON.stringify({
+      op: 'Subscribe',
+      payload: channelName,
+    } as IMiddlewareWebSocketSubscriptionMessage)));
```

Comment:
Using `as` forces TS to think that the data has the defined format. So sometimes wrong formats passes the test. To have the object properly typed I suggest to use this code:
```
const message: IMiddlewareWebSocketSubscriptionMessage = {
      op: 'Subscribe',
      payload: channelName,
};
wsClient.send((JSON.stringify(message));
```